net/FlatBuffers/ByteBuffer.cs

```diff
@@ -63,11 +65,65 @@ public void Reset()
             _pos = 0;
         }
 
+        // Create a new ByteBuffer on the same underlying data.
+        // The new ByteBuffer's position will be 0, but starting
+        // from this ByteBuffer's current position.
+        public virtual ByteBuffer Slice()
+        {
+            return new ByteBufferSlice(_buffer, 0, Position);
+        }
+
+        // Increases the size of the ByteBuffer, and copies the old data towards
+        // the end of the new buffer.
+        public virtual void GrowFront(int newSize)
+        {
+            if ((Length & 0xC0000000) != 0)
+                throw new Exception(
+                    "ByteBuffer: cannot grow buffer beyond 2 gigabytes.");
+
+            if (newSize < Length)
+                throw new Exception("ByteBuffer: cannot truncate buffer.");
+
+            byte[] newBuffer = new byte[newSize];
+            Buffer.BlockCopy(_buffer, 0, newBuffer, newSize - Length,
+                             Length);
+            _buffer = newBuffer;
+        }
+
+        public virtual byte[] ToArray(int pos, int len)
```

Comment:
some methods here are `virtual` and others not.. any reason? I don't think we intend this class to be inherited from, certainly not if it affects performance for basic functions like GetInt etc, which it may.